### PR TITLE
Peggy's first React Native app

### PR DESCRIPTION
**Pull request by pwangy**
My reflections on how this week's project turned out:
This week was terrible. My app does not even begin to work and I'm disappointed in myself. Still been feeling burnt-out and not able to focus and overwhelmed. Help! This project still needs so much more work but I wanted to turn in what I had.
I started by cobbling together different pieces of code from all the code alongs (from both Van and Jennie and the expo tutorial). Knowing that I wanted to use the fan-made Animal Crossing New Horizons API, my idea was to make a simple (haha) random character generator which would trigger on a device being shaken (shook?). I got stuck trying to translate basic react functions to react-native ones. I used an API which doesn't have a randomizer endpoint so I ended up getting blocked with trying to remember how to pull random values from an array. And so my app currently doesn't even load because my mapping function is broken or that's what expo and my console keeps telling me. :( I will keep revisiting this one. But yes, I have styled components and I have multiple pages. And there's a draft called Home in the "production" folder which uses the accelerometer but it hasn't added up to a functioning app yet. 

Things I'd like to have clarified or explained in more detail: How to make time bend and so I can give myself an extra week. But no, nothing. I feel like I have all the information, I just needed to start earlier and work harder and figure it out.

### CODE REVIEW

**Comment by pwangy** [comment]
After a week of continuing to tinker with this, it's finally done! Finally! It is now a mobile app that will load a random Animal Crossing villager on shake.

File: components/DetectShake.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { Accelerometer } from 'expo-sensors'
+
+// Threshold sets shake sensitivity - A lower number gives higher ensitivity. A higher number gives lower sensitivity.
+const threshold = 190
+
+export class DetectShake {
```
Comment (EmOlsz) [inline components/DetectShake.js:6]: Nice usage of outer code ;) I encourage you to deeply investigate how classes work in JavaScript!

File: components/Footer.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import styled from 'styled-components/native'
+
+const FooterContainer = styled.View`
+  margin: 30px 0px 0px 0px;
+  height: 150px;
+  flex-Direction: column-reverse;
```
Comment (EmOlsz) [inline components/Footer.js:7]: I think you meant `flex-direction`

File: components/ProfileSyles.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  width: 98%;
+  margin: 0;
+  padding: 0;
+  justify-content: center;
+  align-items: center;
+`
+
+export const Name = styled.Text`
+width: 100%;
+padding: 0;
```
Comment (EmOlsz) [inline components/ProfileSyles.js:14]: Wrong indentations

File: components/Randomizer.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  let villagerIcon = villager.uri
+
+ // Function to fetch random villager from array
+  const getVillager = () => {
+    const theVillager = VillagerArray[Math.floor(Math.random() * VillagerArray.length)]
+    setVillager(theVillager)
+  }
+
+  // Start = No villager is selected: Instruction text and placeholder image displayed.
+  const [start, setStart] = useState(true)
```
Comment (EmOlsz) [inline components/Randomizer.js:20]: State properties should be grouped together, on the top of each component

File: components/Randomizer.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  const shake = () => {
+    setStart(false)
+  }
+  useEffect(() => {
+    DetectShake.addListener(() => {
+      shake()
+      getVillager()
+      console.log('shaken not stirred')
```
Comment (EmOlsz) [inline components/Randomizer.js:28]: Let's remove this console.log

File: components/Randomizer.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    <>
+      {start ? (
+        <StartState />
+      ) : (
+        <Container>
+        <Image 
+          source={{uri: villagerIcon}}
+          style={{
+            width: 120,
+            height: 120,
```
Comment (EmOlsz) [inline components/Randomizer.js:44]: Those styles should be moved to styled component

File: components/StartState.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+`
+
+export const StartState = () => {
+  return (
+    <Container>
+      <Image 
+        source={placeholder}
+        style={{
+          width: 190,
+          height: 190,
```
Comment (EmOlsz) [inline components/StartState.js:26]: And here as well

**Review by EmOlsz — CHANGES_REQUESTED** [review]

File: components/Footer.js (hunk 2 above)
Comment (pwangy) [reply components/Footer.js:7]: fixed!

File: components/ProfileSyles.js (hunk 3 above)
Comment (pwangy) [reply components/ProfileSyles.js:14]: Ah! Thank you, I missed that.

File: components/Randomizer.js (hunk 4 above)
Comment (pwangy) [reply components/Randomizer.js:20]: Fixed!

File: components/Randomizer.js (hunk 5 above)
Comment (pwangy) [reply components/Randomizer.js:28]: done!

File: components/Randomizer.js (hunk 6 above)
Comment (pwangy) [reply components/Randomizer.js:44]: I tried to move this image styling to ProfileStyles.js but it broke something. :(

File: components/StartState.js (hunk 7 above)
Comment (pwangy) [reply components/StartState.js:26]: I tried to move this image styling to ProfileStyles.js but it broke something. :(

**Comment by pwangy** [comment]
Hi Maks! I made almost all the changes requested. Unfortunately, when I tried to move the image styles into the styles component, it broke something. So, I have left them where they are for now. I hope thats ok!

Also! I found something in the web developer tools in Chrome that let's me simulate a shake, pretty cool!

**Comment by EmOlsz** [comment]
Sure, it's ok, project approved :)

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**